pkg/apis/authorization/types.go

```diff
@@ -17,13 +17,19 @@ limitations under the License.
 package authorization
 
 import (
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
+// +genclient=true
+// +nonNamespaced=true
+// +noMethods=true
+
 // SubjectAccessReview checks whether or not a user or group can perform an action.  Not filling in a
 // spec.namespace means "in all namespaces".
 type SubjectAccessReview struct {
 	unversioned.TypeMeta
+	api.ObjectMeta
```

Comment:
Do they?  I can fix that - you're not quite an object, and we can make an
exception.  You can also implement fake meta accessor and return nothing.

On Thu, Aug 4, 2016 at 3:02 PM, David Eads <notifications@github.com> wrote:

> In pkg/apis/authorization/types.go
> <https://github.com/kubernetes/kubernetes/pull/20573#discussion_r73583095>
> :
>
> >  // SubjectAccessReview checks whether or not a user or group can perform an action.  Not filling in a
> >  // spec.namespace means "in all namespaces".
> >  type SubjectAccessReview struct {
> >  	unversioned.TypeMeta
> > +	api.ObjectMeta
>
> What is kubectl create doing that it requires there to be ObjectMeta?
> What would be the minimal changes to kubectl to be able to handle these
> objects without ObjectMeta?
>
> I feel like these are not "normal" objects because they are not persisted,
> and IIUC, cannot be GET-ed. So, maybe the need a different kind of metadata.
>
> The metadata accessor libraries require everything. In addition, if we use
> this for delegated authorization, it will be a hot-path and including this
> type allows us to take advantage of conforming to the meta.Accessor fast
> path.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20573/files/42b8be284e0c00b2658d26d652153ea70c691626#r73583095>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p9IcwASj_Iqt-7UhRiChop_EMlOVks5qcjc1gaJpZM4HSwZB>
> .
>